weather.py

```diff
@@ -46,7 +55,17 @@ def calculate_mean(weather_data):
     Returns:
         A float representing the mean value.
     """
-    pass
+    for i in range(0, len(weather_data)):
+        weather_data[i] = float(weather_data[i])
+    average = sum(weather_data)/len(weather_data)
```

Comment:
Nice use of the `sum` function for lists. To improve code readability, try to store modified data separately to the original data, e.g. 
```suggestion
    for i in range(0, len(weather_data)):
        float_weather_data[i] = float(weather_data[i])
    average = sum(float_weather_data)/len(weather_data)
```